### PR TITLE
Fix/frontend like dislike update

### PR DESCRIPTION
**Pull request by ivanshamis**

### CODE REVIEW

File: app/Actions/Like/LikePlaceAction.php [hunk 1]
```diff
@@ -51,7 +52,23 @@ public function execute(LikePlaceRequest $request): LikePlaceResponse
         if (!$place) {
             throw new PlaceNotFoundException;
         }
+
+        $liked = LikeStatus::none();
+        $userId = Auth::id();
+        $like = $this->likeRepository->findByUserAndPlace($userId, $request->getPlaceId());
```
Comment (nemoipaha) [inline app/Actions/Like/LikePlaceAction.php]: Same already exists above
` $like = $this->likeRepository->findByUserAndPlace(Auth::id(), $request->getPlaceId());`

File: app/Actions/Like/LikePlaceResponse.php [hunk 2]
```diff
@@ -22,4 +24,9 @@ public function getDislikes()
     {
         return $this->dislikes;
     }
+    
+    public function getLikedStatus()
```
Comment (nemoipaha) [inline app/Actions/Like/LikePlaceResponse.php]: return type

File: app/Actions/Dislike/DislikePlaceAction.php [hunk 3]
```diff
@@ -51,7 +52,23 @@ public function execute(DislikePlaceRequest $request): DislikePlaceResponse
...
             throw new PlaceNotFoundException;
         }
+
+        $liked = LikeStatus::none();
+        $userId = Auth::id();
+        $like = $this->likeRepository->findByUserAndPlace($userId, $request->getPlaceId());
+        if ($like) {
+            $liked = LikeStatus::liked();
+        } else {
+            $dislike = $this->dislikeRepository->findByUserAndPlace($userId, $request->getPlaceId());
```
Comment (nemoipaha) [inline app/Actions/Dislike/DislikePlaceAction.php]: exists above
`$dislike = $this->dislikeRepository->findByUserAndPlace($userId, $request->getPlaceId());`

File: app/Actions/Dislike/DislikePlaceResponse.php [hunk 4]
```diff
@@ -6,11 +6,13 @@ class DislikePlaceResponse
...
 
-    public function __construct(int $likes, int $dislikes)
+    public function __construct(int $likes, int $dislikes, string $likedStatus)
     {
         $this->likes = $likes;
         $this->dislikes = $dislikes;
+        $this->likedStatus = $likedStatus;
     }
 
     public function getLikes()
```
Comment (nemoipaha) [inline app/Actions/Dislike/DislikePlaceResponse.php]: return type

File: app/Actions/Dislike/DislikePlaceResponse.php [hunk 5]
```diff
@@ -22,4 +24,9 @@ public function getDislikes()
     {
         return $this->dislikes;
     }
+    
+    public function getLikedStatus()
```
Comment (nemoipaha) [inline app/Actions/Dislike/DislikePlaceResponse.php]: return type

File: app/Actions/Like/LikePlaceResponse.php [hunk 6]
```diff
@@ -6,11 +6,13 @@ class LikePlaceResponse
...
 
-    public function __construct(int $likes, int $dislikes)
+    public function __construct(int $likes, int $dislikes, string $likedStatus)
     {
         $this->likes = $likes;
         $this->dislikes = $dislikes;
+        $this->likedStatus = $likedStatus;
     }
 
     public function getLikes()
```
Comment (nemoipaha) [inline app/Actions/Like/LikePlaceResponse.php]: return type

File: app/Actions/Like/LikePlaceAction.php (hunk 1 above)
Comment (ivanshamis) [reply app/Actions/Like/LikePlaceAction.php]: this like is after update...

File: app/Http/Controllers/Api/Like/LikeController.php [hunk 7]
```diff
@@ -35,30 +35,31 @@ public function __construct(
...
             );
         } catch (PlaceNotFoundException $exception) {
             return $this->errorResponse('Place not found', 404);
         }
 
-        return $this->successResponse([
-            'likes' => $response->getLikes(),
-            'dislikes' => $response->getDislikes()
-        ], 200);
+        return $this->getLikedPlace($id);
```
Comment (nemoipaha) [inline app/Http/Controllers/Api/Like/LikeController.php]: same as above

File: tests/Feature/Api/Like/LikePlaceApiTest.php [hunk 8]
```diff
@@ -103,7 +103,7 @@ public function testGetLikedNone()
 
         $response->assertStatus(200);
         $response->assertJsonFragment([
-            'liked' => 'NONE'
+            'status' => 'NONE'
```
Comment (nemoipaha) [inline tests/Feature/Api/Like/LikePlaceApiTest.php]: use LikeStatus class

File: tests/Feature/Api/Like/LikePlaceApiTest.php [hunk 9]
```diff
@@ -119,7 +119,7 @@ public function testGetLiked()
 
         $response->assertStatus(200);
         $response->assertJsonFragment([
-            'liked' => 'LIKED'
+            'status' => 'LIKED'
```
Comment (nemoipaha) [inline tests/Feature/Api/Like/LikePlaceApiTest.php]: )

File: tests/Feature/Api/Like/LikePlaceApiTest.php [hunk 10]
```diff
@@ -135,7 +135,7 @@ public function testGetDisliked()
 
         $response->assertStatus(200);
         $response->assertJsonFragment([
-            'liked' => 'DISLIKED'
+            'status' => 'DISLIKED'
```
Comment (nemoipaha) [inline tests/Feature/Api/Like/LikePlaceApiTest.php]: ))

File: app/Http/Controllers/Api/Like/DislikeController.php [hunk 11]
```diff
@@ -8,36 +8,58 @@
...
         try {
-            $response = $this->dislikePlaceAction->execute(
+            $this->dislikePlaceAction->execute(
                 new DislikePlaceRequest($id)
             );
         } catch (PlaceNotFoundException $exception) {
             return $this->errorResponse('Place not found', 404);
         }
 
+        return $this->getLikedPlace($id);
```
Comment (nemoipaha) [inline app/Http/Controllers/Api/Like/DislikeController.php]: controller should return response() only
```
try {
           $this->dislikePlaceAction->execute(
                new DislikePlaceRequest($id)
            );
            
           $placeResponse = $this->getLikedPlaceAction->execute(
                new GetLikedPlaceRequest($id)
            );
        } catch (PlaceNotFoundException $exception) {
            return $this->errorResponse('Place not found', 404);
        } catch (DomainException $exception) {
            return $this->errorResponse($exception->getMessage(), 400);
        }
```

File: tests/Feature/Api/Like/LikePlaceApiTest.php [hunk 12]
```diff
@@ -103,7 +103,7 @@ public function testGetLikedNone()
 
         $response->assertStatus(200);
         $response->assertJsonFragment([
-            'liked' => 'NONE'
+            'likeStatus' => 'NONE'
```
Comment (nemoipaha) [inline tests/Feature/Api/Like/LikePlaceApiTest.php]: LikeStatus class

File: tests/Feature/Api/Like/LikePlaceApiTest.php [hunk 13]
```diff
@@ -119,7 +119,7 @@ public function testGetLiked()
 
         $response->assertStatus(200);
         $response->assertJsonFragment([
-            'liked' => 'LIKED'
+            'likeStatus' => 'LIKED'
```
Comment (nemoipaha) [inline tests/Feature/Api/Like/LikePlaceApiTest.php]: same

**Pull request merged**